source/common/http/header_map_impl.cc

```diff
@@ -304,6 +293,22 @@ HeaderMapImpl::HeaderMapImpl(
   }
 }
 
+void HeaderMapImpl::copyFrom(const HeaderMap& header_map) {
+  header_map.iterate(
+      [](const HeaderEntry& header, void* context) -> HeaderMap::Iterate {
+        // TODO(mattklein123) PERF: Avoid copying here is not necessary.
+        HeaderString key_string;
+        key_string.setCopy(header.key().c_str(), header.key().size());
+        HeaderString value_string;
+        value_string.setCopy(header.value().c_str(), header.value().size());
```

Comment:
Best practice is to use .data() rather than .c_str() here (x2), although I admit in practice I don't see how they can differ :)

Reply:
There is no `.data()` on `header.value()`.